netty/src/main/java/io/grpc/transport/netty/ProtocolNegotiators.java

```diff
@@ -181,8 +183,13 @@ public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
     }
 
     @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+      fail(ctx, cause);
+    }
+
+    @Override
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
-      failBufferedAndClose(ctx);
+      fail(ctx, new Exception("Connection broken while performing protocol negotiation"));
```

Comment:
channelInactive. UNAVAILABLE seems reasonable.

close. No, not cancelled, since even if we willingly closed the connection that would just make it UNAVAILABLE. INTERNAL or UNAVAILABLE sounds fine to me. If in real doubt we can do UNKNOWN.

http2 plaintext upgrade failed. No, not FAILED_PRECONDITION; UNAVAILABLE is strictly better here (see the litmus test in FAILED_PRECONDITION's doc). UNAVAILABLE does seem most appropriate, since I think that is the same status we should give (but we may not be doing so) if hostname lookup fails or can't create a TCP connection.